<a href="https://colab.research.google.com/github/soustan/Internship-Data-Analitic/blob/main/Internship_part_4.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [4]:
import pandas as pd
import numpy as np

In [7]:
#Load the region roles
from google.colab import files
upload = files.upload()

Saving region_roles_26_05_22.xlsx to region_roles_26_05_22.xlsx


In [8]:
reg_rol = pd.read_excel('region_roles_26_05_22.xlsx')

In [12]:
reg_rol.head(5)

Unnamed: 0        id  premium  \
0           0  49059633    False   
1           1  54653519    False   
2           2  55116066    False   
3           3  54031261    False   
4           4  55243040    False   

                                               name  \
0           Младший специалист ручного тестирования   
1                                   Тестировщик.Net   
2                                         Пентестер   
3  Ведущий Специалист по информационным технологиям   
4                   Специалист по защите информации   

                                          department  has_test  \
0                                                NaN     False   
1                                                NaN     False   
2                                                NaN     False   
3  {'id': 'pochta-4352-office', 'name': '«Почта Р...     False   
4                                                NaN     False   

   response_letter_required  \
0                     False   
1                     False   
2                     False   
3                     False   
4                     False   

                                                area  \
0  {'id': '15', 'name': 'Астрахань', 'url': 'http...   
1  {'id': '15', 'name': 'Астрахань', 'url': 'http...   
2  {'id': '15', 'name': 'Астрахань', 'url': 'http...   
3  {'id': '15', 'name': 'Астрахань', 'url': 'http...   
4  {'id': '15', 'name': 'Астрахань', 'url': 'http...   

                                              salary  \
0                                                NaN   
1  {'from': 25000, 'to': 50000, 'currency': 'RUR'...   
2                                                NaN   
3  {'from': 26000, 'to': None, 'currency': 'RUR',...   
4  {'from': 20000, 'to': 20000, 'currency': 'RUR'...   

                                 type  ... working_days  \
0  {'id': 'open', 'name': 'Открытая'}  ...           []   
1  {'id': 'open', 'name': 'Открытая'}  ...           []   
2  {'id': 'open', 'name': 'Открытая'}  ...           []   
3  {'id': 'open', 'name': 'Открытая'}  ...           []   
4  {'id': 'open', 'name': 'Открытая'}  ...           []   

  working_time_intervals  working_time_modes accept_temporary  \
0                     []                  []            False   
1                     []                  []            False   
2                     []                  []            False   
3                     []                  []            False   
4                     []                  []            False   

                                         description  \
0  <p>В некоторых проектах Яндексу помогают компа...   
1  <p><em><strong>Мы ищем талантливого и амбициоз...   
2  <p>Лидер на рынке систем информационной безопа...   
3  <ul> <li>Администрирование компьютерной сети</...   
4  <strong>Обязанности:</strong> <ul> <li>Обеспеч...   

                                          experience  \
0        {'id': 'noExperience', 'name': 'Нет опыта'}   
1  {'id': 'between1And3', 'name': 'От 1 года до 3...   
2    {'id': 'between3And6', 'name': 'От 3 до 6 лет'}   
3        {'id': 'noExperience', 'name': 'Нет опыта'}   
4        {'id': 'noExperience', 'name': 'Нет опыта'}   

                                          key_skills  \
0                                                 []   
1  [{'name': 'Git'}, {'name': 'SQL'}, {'name': 'Т...   
2  [{'name': 'SQL'}, {'name': 'Linux'}, {'name': ...   
3  [{'name': 'Пользователь ПК'}, {'name': 'Работа...   
4  [{'name': 'MS Dos'}, {'name': 'Диагностика ПК'...   

                                     specializations                region  \
0  [{'id': '1.172', 'name': 'Начальный уровень, М...  Астраханская область   
1  [{'id': '1.221', 'name': 'Программирование, Ра...  Астраханская область   
2  [{'id': '1.221', 'name': 'Программирование, Ра...  Астраханская область   
3  [{'id': '1.110', 'name': 'Компьютерная безопас...  Астраханская область   
4  [{'id': '1.110', 'name': 'Компьютерная безопас.

In [46]:
reg_rol['salary'].str.split('\n')[0]

nan

In [55]:
new_reg_rol = reg_rol['salary'].str.split('\n',expand=True)
new_reg_rol

0
0                                                    NaN
1      {'from': 25000, 'to': 50000, 'currency': 'RUR'...
2                                                    NaN
3      {'from': 26000, 'to': None, 'currency': 'RUR',...
4      {'from': 20000, 'to': 20000, 'currency': 'RUR'...
...                                                  ...
69651                                                NaN
69652  {'from': 130000, 'to': None, 'currency': 'RUR'...
69653  {'from': 50000, 'to': None, 'currency': 'RUR',...
69654  {'from': 85000, 'to': None, 'currency': 'RUR',...
69655                                                NaN

[69656 rows x 1 columns]

In [ ]:
#new_reg_rol.columns = ['From', 'to', 'Currency']
#splited_items = [a.split('=') for a in new_reg_rol[0]]
#new_reg_rol

In [ ]:
new_reg_rol = reg_rol['salary'].str.split('\n').\
apply(lambda x: {item[0]: item[1] for item in [a.split('=') for a in x]}).\
apply(pd.Series)

In [ ]:
reg_rol['salary_1'] = reg_rol['salary'].map(lambda x:x. split('|')[0])
reg_rol['salary_2'] = reg_rol['salary'].map(lambda x:x. split('|')[1])
reg_rol['salary_3'] = reg_rol['salary'].map(lambda x:x. split('|')[2])

In [ ]:
reg_rol

In [32]:
reg_rol.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 69656 entries, 0 to 69655
Data columns (total 34 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   Unnamed: 0                69656 non-null  int64  
 1   id                        69656 non-null  int64  
 2   premium                   69656 non-null  bool   
 3   name                      69656 non-null  object 
 4   department                5263 non-null   object 
 5   has_test                  69656 non-null  bool   
 6   response_letter_required  69656 non-null  bool   
 7   area                      69656 non-null  object 
 8   salary                    37590 non-null  object 
 9   type                      69656 non-null  object 
 10  address                   40318 non-null  object 
 11  response_url              1 non-null      object 
 12  sort_point_distance       0 non-null      float64
 13  published_at              69656 non-null  object 
 14  create

В ходе изучения Датасет, удалось оценить его размер, который равен 15.7 MB. Количество строк в датасет - 69656. Количество колонок(столбцов) - 34. В результате чего, стало понятным, что:
Количество булевых строк равно 5, Строк с плавающей запятой равно 1, строк с целым числом равно 2, строк с типом данных object равно 26. Столбцы, с именами id, premium, name, department, type, address и другие, имеют пустые или нулевые значения. В результате, из данных подлежащих анализу. имеются значения: Целые числа, значения отличные от нуля, а так же значения типа object, столбец salary.

In [38]:
reg_rol = reg_rol.astype'salary': np.int})

/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:1: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
  """Entry point for launching an IPython kernel.


ValueError: ignored

In [2]:
#Load the result roles
from google.colab import files
upload = files.upload()

Saving result roles (16.06.2022).xlsx to result roles (16.06.2022).xlsx


In [23]:
res_rol = pd.read_excel('result_roles_16_06_2022.xlsx')

In [24]:
res_rol.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 76382 entries, 0 to 76381
Data columns (total 51 columns):
 #   Column                    Non-Null Count  Dtype 
---  ------                    --------------  ----- 
 0   Unnamed: 0                76382 non-null  int64 
 1   id                        76382 non-null  int64 
 2   premium                   76382 non-null  bool  
 3   name                      76382 non-null  object
 4   department                4902 non-null   object
 5   has_test                  76380 non-null  object
 6   response_letter_required  76382 non-null  object
 7   area                      76381 non-null  object
 8   salary                    40874 non-null  object
 9   type                      76379 non-null  object
 10  published_at              76380 non-null  object
 11  created_at                76381 non-null  object
 12  archived                  76380 non-null  object
 13  insider_interview         570 non-null    object
 14  url                   